avivator/src/components/Controller/components/GlobalSelectionSlider.js

```diff
@@ -1,18 +1,55 @@
 import React from 'react';
 import Grid from '@material-ui/core/Grid';
 import Slider from '@material-ui/core/Slider';
+import debounce from 'lodash/debounce';
 import { range, getMultiSelectionStats } from '../../../utils';
 import {
   useChannelSettings,
   useChannelSetters,
-  useViewerStore
+  useViewerStore,
+  useImageSettingsStore
 } from '../../../state';
 
 export default function GlobalSelectionSlider(props) {
   const { size, label } = props;
   const { setPropertiesForChannels } = useChannelSetters();
   const { selections, loader } = useChannelSettings();
   const { setViewerState, globalSelection } = useViewerStore();
+  const { setImageSetting } = useImageSettingsStore();
+  const changeSelection = debounce(
+    (event, newValue) => {
+      setViewerState({
+        isChannelLoading: selections.map(i => true)
+      });
+      const newSelections = [...selections].map(sel => ({
+        ...sel,
+        [label]: newValue
+      }));
+      getMultiSelectionStats({
+        loader,
+        selections: newSelections,
+        use3d: false
+      }).then(({ domains, sliders }) => {
+        setImageSetting({
+          onViewportLoad: () => {
+            setPropertiesForChannels(range(newSelections.length), {
+              domains,
+              sliders
+            });
+            setImageSetting({ onViewportLoad: () => {} });
+            setViewerState({
+              isChannelLoading: selections.map(i => false)
+            });
+          }
+        });
```

Comment:
Similar to the comment in the other PR, is it possible to remove the "multi" variants of these settings. It would be easier to use them in a modular manner. In addition, currently we need to wait for all the "channel" stats before updating any of the UI. Why not fire off all these changes asynchronously? Update each domain & slider independently, whenvever the data comes back.

Reply:
> Similar to the comment in the other PR, is it possible to remove the "multi" variants of these settings. 

You mean the multiple channel version?  Sure I can just use the single channel version and remove the other.

>In addition, currently we need to wait for all the "channel" stats before updating any of the UI. Why not fire off all these changes asynchronously? Update each domain & slider independently, whenvever the data comes back.

The idea here is that the channel stats come back much much quicker than the actual data often times for 3d volumes so we only want the sliders change once the 3d volume is loaded and ready for rendering.  Otherwise you get a sort of flashing effect where the sliders change but the actual data has not.